src/DateRangePicker.jsx

```diff
@@ -90,11 +90,17 @@ const DateRangePicker = React.createClass({
   componentWillReceiveProps(nextProps) {
     var nextDateStates = this.getDateStates(nextProps);
     var nextEnabledRange = this.getEnabledRange(nextProps);
-
-    this.setState({
+    var updatedState = {
       dateStates: this.state.dateStates && Immutable.is(this.state.dateStates, nextDateStates) ? this.state.dateStates : nextDateStates,
       enabledRange: this.state.enabledRange && this.state.enabledRange.isSame(nextEnabledRange) ? this.state.enabledRange : nextEnabledRange,
-    });
+    };
+
+    if (nextProps.value ){
```

Comment:
Going to review this change a bit more; I'm not sure if this works correctly when multiple dates are selected